crawl-ref/source/monster.cc

```diff
@@ -1955,8 +1955,7 @@ bool monster::pickup_missile(item_def &item, bool msg, bool force)
                 // Don't drop huge stacks for tiny stacks.
                 if (item.launched_by(*launch)
                     && (!miss->launched_by(*launch)
-                        || item.sub_type == MI_SLING_BULLET
-                           && miss->sub_type == MI_STONE
```

Comment:
You can remove the MI_STONE line here as well; it will now be covered by the other checks in this `if` statement.